test/simulation-test.coffee

```diff
@@ -66,237 +66,125 @@ describe "Simulation", ->
-            duration: 10
-            timeStep: 1
-            result:   -1
-          }
-          # odder examples...
-          {
-            startA:   10
-            startB:   0
-            formula:  "in * 2"
-            duration: 10
-            timeStep: 0.2
-            result:   4
-          }
-          {
-            startA:   10
-            startB:   20
-            formula:  "in * 2"
-            duration: 10
-            timeStep: 0.2
-            result:   4
-          }
-          {
-            startA:   10
-            startB:   1
-            formula:  "out + out"
-            duration: 8
-            timeStep: 1
-            result:   256
-          }
-          {
-            startA:   10
-            startB:   1
-            formula:  "out + out"
-            duration: 4
-            timeStep: 0.5
-            result:   1
-          }
-        ]
-      describe "each scenario", ->
-        it "should compute correctly", ->
-          _.each @scenarios, (scenario) ->
-            nodeA = new Node({initialValue: scenario.startA})
-            nodeB = new Node({initialValue: scenario.startB, isAccumulator: scenario.bIsAccumulator})
-            LinkNodes(nodeA, nodeB, scenario.formula)
+    # We can describe each scenario as an object:
+    # Each single-letter key is a node. Values can be a number (the initial value
+    #   for independent variables), a string "x+" (the initial value for collectors),
+    #   or null (dependent variables).
+    # Each two-letter node is a link, with the formula for the link.
+    # Results is an array of arbitrary length, describing the expected result for each
+    # node on each step.
+    describe "for other scenarios", ->
+      scenarios = [
+        # cascade independent and dependent variables (A->B->C)
+        {A: 50, B: null, C: null, AB: "1 * in", BC: "0.1 * in",
+        results: [
+            [50, 50, 5]
+            [50, 50, 5]
+        ]}
+
+        # cascade independent and dependent variables with negative relationship (A->B->C)
+        {A: 50, B: null, C: null, AB: "0.1 * in", BC: "-1 * in",
+        results: [
+            [50, 5, -5]
+            [50, 5, -5]
+        ]}
+
+        # basic collector (A->[B])
+        {A:5, B:"50+", AB: "1 * in",
+        results: [
+          [5, 50]
+          [5, 55]
+        ]}
+
+        # basic collector with feedback (A<->[B])
+        {A:null, B:"50+", AB: "1 * in", BA: "1 * in",
+        results: [
+          [50, 50]
+          [100, 100]
+          [200, 200]
+        ]}
+
+        # three-node graph (>-) with averaging
+        {A: 10, B: 20, C: null, AC: "1 * in", BC: "1 * in",
+        results: [
+            [10, 20, 15]
+            [10, 20, 15]
+        ]}
+
+        # three-node graph (>-) with non-linear averaging
+        {A: 10, B: 20, C: null, AC: "1 * in", BC: "0.1 * in",
+        results: [
+            [10, 20, 6]
+            [10, 20, 6]
+        ]}
+
+        # three-node graph with collector (>-[C])
```

Comment:
Just a note: I find it interesting that dependent variables will average their inputs, while an accumulator will sum. It makes sense, but it gave me pause.